pkg/provider/azure_utils.go

```diff
@@ -300,3 +300,16 @@ func sameContentInSlices(s1 []string, s2 []string) bool {
 	}
 	return true
 }
+
+func removeDuplicatedSecurityRules(rules []network.SecurityRule) []network.SecurityRule {
+	ruleNames := make(map[string]bool)
+	for i := len(rules) - 1; i >= 0; i-- {
+		if _, ok := ruleNames[to.String(rules[i].Name)]; ok {
+			klog.Warningf("Found duplicated rule %s, will be removed.", to.String(rules[i].Name))
+			rules = append(rules[:i], rules[i+1:]...)
+		} else {
+			ruleNames[to.String(rules[i].Name)] = true
```

Comment:
I prefer not put this line in else. Fewer indention.

Reply:
done